registration/include/pcl/registration/impl/ndt.hpp

```diff
@@ -124,30 +121,30 @@ NormalDistributionsTransform<PointSource, PointTarget>::computeTransformation (P
     // Solve for decent direction using newton method, line 23 in Algorithm 2 [Magnusson 2009]
     Eigen::JacobiSVD<Eigen::Matrix<double, 6, 6> > sv (hessian, Eigen::ComputeFullU | Eigen::ComputeFullV);
     // Negative for maximization as opposed to minimization
-    delta_p = sv.solve (-score_gradient);
+    Eigen::Matrix<double, 6, 1> delta = sv.solve (-score_gradient);
 
     //Calculate step length with guarnteed sufficient decrease [More, Thuente 1994]
-    double delta_p_norm = delta_p.norm ();
+    double delta_norm = delta.norm ();
 
-    if (delta_p_norm == 0 || std::isnan(delta_p_norm))
+    if (delta_norm == 0 || std::isnan(delta_norm))
     {
-      trans_probability_ = score / static_cast<double> (input_->points.size ());
-      converged_ = delta_p_norm == delta_p_norm;
+      trans_probability_ = score / static_cast<double> (input_->size ());
+      converged_ = delta_norm == 0;
       return;
     }
 
-    delta_p.normalize ();
-    delta_p_norm = computeStepLengthMT (p, delta_p, delta_p_norm, step_size_, transformation_epsilon_ / 2, score, score_gradient, hessian, output);
-    delta_p *= delta_p_norm;
+    delta.normalize ();
```

Comment:
Divide by `delta_norm` instead? Might be faster

Reply:
Done